src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java

```diff
@@ -38,75 +52,259 @@
+  }
+
+  public ChatAssistant(List<Message> messages) {
+    this.messages = messages;
+    content.getElement().setAttribute("slot", "content");
+    content.setItems(messages);
+
+    content.setRenderer(new ComponentRenderer<ChatMessage, Message>(
+        message -> new ChatMessage(message), (component, message) -> {
+          ((ChatMessage) component).setMessage(message);
+          return component;
+        }));
+    this.add(content);
+    messageInput = new MessageInput();
+    messageInput.setSizeFull();
+    defaultSubmitListenerRegistration = messageInput
+        .addSubmitListener(se -> this.sendMessage(Message.builder().messageTime(LocalDateTime.now())
+            .name("User").content(se.getValue()).build()));
+    whoIsTyping = new Span();
+    whoIsTyping.setClassName("chat-assistant-who-is-typing");
+    whoIsTyping.setVisible(false);
+    VerticalLayout vl = new VerticalLayout(whoIsTyping, messageInput);
+    vl.setSpacing(false);
+    vl.setMargin(false);
+    vl.setPadding(false);
+    this.setFooterComponent(vl);
+    this.getElement().addEventListener("bot-button-clicked", this::handleClick).addEventData("event.detail");
+    
+    Icon minimize = VaadinIcon.CHEVRON_DOWN_SMALL.create();
+    minimize.addClickListener(ev -> this.setMinimized(!minimized));
+    Span title = new Span("Chat Assistant");
+    title.setWidthFull();
+    HorizontalLayout headerBar = new HorizontalLayout(title, minimize);
+    headerBar.setWidthFull();
+    this.setHeaderComponent(headerBar);
+  }
+
+  private void handleClick(DomEvent event) {
+    minimized = event.getEventData().getObject("event.detail").getBoolean("minimized");
+    if (!minimized) {
+      refreshContent();
+    }
+  }
+  
+  /**
+   * Sets the data provider of the internal VirtualList
+   * @param dataProvider
+   */
+  public void setDataProvider(DataProvider<Message, ?> dataProvider) {
+    content.setDataProvider(dataProvider);
+  }
 
   /**
-   * Sends a message represented by the string message programmatically to the component, with
-   * default settings.
-   *
-   * @param message
+   * Uses the provided string as the text shown over the message input to indicate that someone is typing
+   * @param whoIsTyping
+   */
+  public void setWhoIsTyping(String whoIsTyping) {
+    this.whoIsTyping.setText(whoIsTyping);
+    this.whoIsTyping.setVisible(true);
+  }
+  
+  /**
+   * Returns the current text shown over the message input to indicate that someone is typing
+   * @return the current text or null if not configured
+   */
+  public String getWhoIsTyping() {
+    return whoIsTyping.getText();
+  }
+  
+  /**
+   * Clears the text shown over the message input to indicate that someone is typing
+   */
+  public void clearWhoIsTyping() {
+    this.whoIsTyping.setText(null);
+    this.whoIsTyping.setVisible(false);
+  }
+  
+  /**
+   * Sets the SubmitListener that will be notified when the user submits a message on the underlying messageInput
```

Comment:
Description should end with a dot. There should be a empty line between the description and the tags. Param and return tags should have a brief description also, and shouldn't end with a dot. See documentation https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#format